auslib/test/web/test_client.py

```diff
@@ -919,22 +920,23 @@ def testEmptySnippetOn404(self):
     def testEmptySnippetOn500(self):
         with mock.patch('auslib.web.views.client.ClientRequestView.get') as m:
             m.side_effect = Exception('I break!')
-            ret = self.client.get('/update/4/b/1.0/1/p/l/a/a/a/a/1/update.xml')
-            self.assertUpdatesAreEmpty(ret)
+            with self.assertRaises(Exception) as exc:
+                self.client.get('/update/4/b/1.0/1/p/l/a/a/a/a/1/update.xml')
+                self.assertEqual('I break!', exc.exception.value())
```

Comment:
I didn't know you could use this as a context manager - neat!

However, I don't think this assertEqual ever runs. It looks to me like nothing in the block will run after the Exception is raised. The example on https://docs.python.org/2/library/unittest.html does some verification of the exception outside the block, like this:
```
with self.assertRaises(SomeException) as cm:
    do_something()

the_exception = cm.exception
self.assertEqual(the_exception.error_code, 3)
```